src/bridge/cockpitcpusamples.c

```diff
@@ -112,3 +112,72 @@ cockpit_cpu_samples (CockpitSamples *samples)
   g_strfreev (lines);
   g_free (contents);
 }
+
+void
+cockpit_cpu_temperature (CockpitSamples *samples)
+{
+  for (int i = 0; TRUE; i++)
+    {
+      gsize len;
+      g_autoptr (GError) error = NULL;
+      g_autofree gchar *path = g_strdup_printf ("/sys/class/hwmon/hwmon%d/name", i);
+      g_autofree gchar *name = NULL;
+
+      if (!g_file_get_contents (path, &name, &len, &error))
+        {
+          if (!g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
+              g_message ("error loading name: %s", error->message);
+
+          break;
+        }
+
+      g_strchomp(name);
+
+      if (!g_str_equal (name, "coretemp") && !g_str_equal (name, "cpu_thermal") &&
```

Comment:
Comment explaining this, please :)

Reply:
Or what about doing the "filtering" in JavaScript and pass all temperature values to the client?